pkg/operator/controller/certificate/controller.go

```diff
@@ -105,6 +107,48 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
 			if _, err := r.ensureDefaultCertificateForIngress(ca, deployment.Namespace, deploymentRef, ingress); err != nil {
 				errs = append(errs, fmt.Errorf("failed to ensure default cert for %s: %v", ingress.Name, err))
 			}
+			// The ingress canary verifies that the default ingress controller is functioning. Since it uses a
+			// passthrough route, we mirror the default ingress controller's certificate in the canary to detect any
+			// issues with that certificate, so if the default controller's cert is updated, update the canary's cert as
+			// well.
+			if ingress.Name == manifests.DefaultIngressControllerName {
+				log.Info("Ensuring canary certificate")
+				daemonset := &appsv1.DaemonSet{}
+				err = r.client.Get(ctx, controller.CanaryDaemonSetName(), daemonset)
+				if err != nil {
+					if errors.IsNotFound(err) {
+						// All ingresses should have a deployment, so this one may not have been
+						// created yet. Retry after a reasonable amount of time.
+						log.Info("Canary daemonset not found; will retry default cert sync")
+						result.RequeueAfter = 5 * time.Second
+					} else {
+						errs = append(errs, fmt.Errorf("failed to get daemonset: %w", err))
+					}
+				} else {
+					defaultCertName := controller.RouterEffectiveDefaultCertificateSecretName(ingress, deployment.Namespace)
+					defaultCert := &corev1.Secret{}
+					if err := r.client.Get(ctx, defaultCertName, defaultCert); err != nil {
+						errs = append(errs, fmt.Errorf("failed to get certificate for canary: %w", err))
+					}
+					canaryCert := defaultCert.DeepCopy()
+					canaryCertName := controller.RouterEffectiveDefaultCertificateSecretName(ingress, daemonset.Namespace)
```

Comment:
This should work, but is there a reason to copy the name from the effective certificate?  If you used a static name, you could get rid of `canarySecretName` in `pkg/operator/controller/canary/daemonset.go` and simplify the logic a bit.

Reply:
Well...  I wonder whether the issue referenced at https://github.com/openshift/cluster-ingress-operator/pull/1155#discussion_r2591642782 will prevent garbage collection from working properly?